internal/k8s/secrets/validation.go

```diff
@@ -109,6 +112,23 @@ func ValidateOIDCSecret(secret *api_v1.Secret) error {
 	return nil
 }
 
+// ValidateAPIKeySecret validates the secret. If it is valid, the function returns nil.
+func ValidateAPIKeySecret(secret *api_v1.Secret) error {
+	if secret.Type != SecretTypeAPIKey {
+		return fmt.Errorf("APIKey secret must be of the type %v", SecretTypeAPIKey)
+	}
+
+	uniqueKeys := make(map[string]bool)
+	for _, key := range secret.Data {
+		if uniqueKeys[string(key)] {
+			return fmt.Errorf("API Keys cannot be repeated")
+		}
+		uniqueKeys[string(key)] = true
+	}
+
+	return nil
```

Comment:
Might be worth validating that uniqueKeys has at least one entry?

Reply:
Enforcing uniqueKeys having at least one entry would mean we will reject an empty kubernetes secrets for API key policy. We should allow users to supply a secret without any client/api key combination. if we reject empty secrets, the policy would become invalid and returns 500 for all requests. 

We think an API policy without any client/API key is a legit use case. If user deploys a policy with empty secret, the policy would not become invalid, and the map in the nginx config would become empty, therefore nginx would return 403 for all requests using the specified headers or queries, the same behavior as having the wrong API key in the request. Users can add clients/api keys to the secret without changing the policy or virtual server.